src/storage.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python3
+
+# Copyright 2024 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Handle Juju Storage Events."""
+
+import logging
+from socket import gethostname
+from subprocess import CalledProcessError
+
+from charms.operator_libs_linux.v2.snap import SnapCache
+from ops.charm import (
+    CharmBase,
+    EventBase,
+    EventSource,
+    ObjectEvents,
+    StorageAttachedEvent,
+    StorageDetachingEvent,
+)
+from ops.framework import Object, StoredState
+
+import microceph
+
+logger = logging.getLogger(__name__)
+
+
+class StorageUpdatedEvent(EventBase):
+    """Storage Event for the charm to observe."""
+
+    def __init__(self, handle):
+        super().__init__(handle)
+
+
+class MicroCephStorageEvents(ObjectEvents):
+    """Events related to MicroCluster apps."""
+
+    storage_updated = EventSource(StorageUpdatedEvent)
+
+
+class StorageHandler(Object):
+    """The Storage class manages the Juju storage events.
+
+    Observes the following events:
+    1) *_storage_attached
+    2) *_storage_detaching
+    """
+
+    on = MicroCephStorageEvents()
+    charm = None
+    _stored = StoredState()
+
+    def __init__(self, charm: CharmBase, name="storage"):
+        super().__init__(charm, name)
+        self._stored.set_default(disk=[], wal=[], db=[])
+        self.charm = charm
+
+        logger.info("TEST: INITIALISING STORAGE HANDLER CLASS.")
+
+        # Auto disk handler.
+        self.framework.observe(
+            getattr(charm.on, "plaindisk_storage_attached"), self._on_plaindisk_attached
```

Comment:
do you need to use `getattr`? I was under the impression that simply using `charm.on.plaindisk_storage_attached` was enough.